### PR TITLE
Fix pending integration test by moving postTx outside of the atomically block

### PR DESCRIPTION
**Pull request by Unisay**
- [x] I have re-enabled pending test `STAKE_POOLS_JOIN_05`
- [x] Moved `postTx` out of the `atomically` block, this makes test suite pass.

### Issue Number

ADP-2662

### CODE REVIEW

**Comment by Unisay** [comment]
bors try

**Comment by Unisay** [comment]
bors try

**Comment by Unisay** [comment]
bors try

**Comment by Unisay** [comment]
bors try

File: lib/wallet/integration/src/Test/Integration/Faucet.hs [hunk 1]
```diff
@@ -2418,6 +2418,7 @@ shelleyIntegrationTestFunds = mconcat
     preregKeyWalletFunds = map ((,defaultAmt) . Address . unsafeFromHex)
         [ "6199a7c32aaa55a628d936b539f01d5415318dec8bcb5e59ec71af695b"
         , "61386c7a86d8844f4085a50241556043c9842d72c315c897a42a8a0510"
+        , "614adf026163a9182cd4087acc966aef46e37474a2994a06a69de44d6f"
```
Comment (Anviking) [inline lib/wallet/integration/src/Test/Integration/Faucet.hs]: If we rely on having enough spare UTxOs for failing pending txs not to consume them all, I think we should add more.

Actually, we might as well not hard-code the addresses, but use the same approach as for the `seqMnemonics`. We could

1. Define `preregStakeKeyWallet :: Mnemonic 15`
2. Import it from the test itself (optional, but why not) https://github.com/input-output-hk/cardano-wallet/blob/56784ab870f827d223c3dd129d3eb5cc23cad3a5/lib/wallet/integration/src/Test/Integration/Scenario/API/Shelley/Migrations.hs#L593
3. Generate 10 addresses automatically like this https://github.com/input-output-hk/cardano-wallet/blob/56784ab870f827d223c3dd129d3eb5cc23cad3a5/lib/wallet/integration/src/Test/Integration/Faucet.hs#L2366, but with the `preregStakeKeyWallet` mnemonic.

File: lib/wallet/src/Cardano/Wallet.hs [hunk 2]
```diff
@@ -1949,17 +1950,19 @@ buildSignSubmitTransaction ti db@DBLayer{..} netLayer txLayer pwd walletId
...
-                & fmap (BuiltTx{..},)
-                & liftIO
+        postTx netLayer builtSealedTx
+            & throwWrappedErr wrapNetworkError
+            & liftIO
+
+        slotToUTCTime slot
+            & interpretQuery (neverFails "slot is ahead of the node tip" ti)
+            & fmap (BuiltTx{..},)
+            & liftIO
```
Comment (Anviking) [inline lib/wallet/src/Cardano/Wallet.hs:1965]: This is now not needed to make `JOIN_05` pass? Regardless, this seems sane to me for the sake 1) not doing potentially slow networking in `atomically` 2) still updating the `pending` set and the change state together, even though I'm not sure of the exact implications 🤔

File: lib/wallet/integration/src/Test/Integration/Faucet.hs (hunk 1 above)
Comment (Unisay) [reply lib/wallet/integration/src/Test/Integration/Faucet.hs]: Done

**Review by Anviking — APPROVED** [review]

**Comment by Unisay** [comment]
bors merge

**Pull request merged**